File: Programa.pas
program Programa;

var
  meuArquivo: File;
  umByte    : Byte;

begin
  AssignFile(meuArquivo, 'c:\pasta\NomeArquivo.bmp');
  Reset(meuArquivo, 1);
  BlockRead(meuArquivo, umByte, 1);
  CloseFile(meuArquivo);

  case umByte of
    137:
      WriteLn('Formato PNG');
    255:
      WriteLn('Formato Jpeg');
    73:
      WriteLn('Formato TIFF');
    66:
      WriteLn('Formato Bitmap');
    71:
      WriteLn('Formato GIF');
  else
    WriteLn('Formato desconhecido');
  end;

end.

File: programa.php
<?php
$buffer = file_get_contents("a.jpg");

if (!$buffer) {
  die("Erro ao ler arquivo\n");
}

$primeiro_byte = ord($buffer[0]);

switch ($primeiro_byte) {
  case 137:
    echo("Formato PNG");
    break;
 
  case 255:
    echo("Formato JPEG");
    break;
  
  case 73:
    echo("Formato TIFF");
    break;
  
  case 66:
    echo("Formato Windows Bitmap");
    break;
  
  case 71:
    echo("Formato GIF");
    break;       

  default:
    echo("Formato desconhecido");
}